semcore/ellipsis/src/Ellipsis.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { RefObject, useRef, useMemo, useState, useLayoutEffect } from 'react';
+import createComponent, { Component, sstyled } from '@semcore/core';
+import Tooltip from '@semcore/tooltip';
+import { Flex } from '@semcore/flex-box';
+import { useResizeObserver } from './useResizeObserver';
+
+import style from './style/ellipsis.shadow.css';
+import reactToText from '@semcore/utils/lib/reactToText';
+import getOriginChildren from '@semcore/utils/lib/getOriginChildren';
+
+type AsProps = {
+  maxline?: number;
+  trim?: 'end' | 'middle';
+  tooltip?: string;
+  styles?: React.CSSProperties;
+  resizeObserver?: { width: number };
```

Comment:
After `resizeObserver` becoming not a special object with methods to subscribe and unsubscribe from changing size I suppose that essence of `resizeObserver` prop changed enough to change the name of that prop. To something like `containerRect`